### DISCUSSION TITLE
What's the SOTA architecture/framework for retrieval-based open-domain chatbots?

### DISCUSSION THREAD

**Discussion by Lisennk** [facebookresearch/ParlAI#4294] [Q&A] [answered]
Hi all. What is the go-to solution for creating a retrieval-based open-domain chatbot in 2022? We have a huge dataset, but it's not labeled to use RASA or other intent-matching engines.

**Comment by stephenroller** [comment, marked as answer]
Retrieval models have fallen out of fashion compared to generative models recently, but AFAIK the poly-encoder model remains very strong.

**Reply by Lisennk** [reply]
Thanks, Stephen. I thought that it makes sense to get retrieval and generative models output to pass it to the reranker and that way choose what's best. What do you think about this approach compared to just using generative models?

**Reply by stephenroller** [reply]
It's reasonable, but if your re-ranker is substantially less powerful (smaller, less trained, etc) than the generator, it may have a hard time distinguishing.
